vignettes/profiling.Rmd

```diff
@@ -0,0 +1,68 @@
+---
+title: "Profiling Performance"
+author: "Thomas Lin Pedersen"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Profiling Performance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The performance of `gtable` is critical as it has the potential to slow down
+plot creation a lot. In order to continuously monitor the performance of gtable
+the following piece of code is used to generate a profiling and inspect it:
+
+```{r}
+library(ggplot2)
+library(profvis)
+
+p <- ggplot() + 
+  scale_x_continuous(limits = c(0, 1)) + 
+  scale_y_continuous(limits = c(0, 1))
+
+p_build <- ggplot_build(p)
+
+profile <- profvis(for (i in seq_len(100)) ggplot_gtable(p_build))
+
+profile
```

Comment:
Shouldn't this be load from disk? So e.g. pkgdown will generate the same plot that you see?

Reply:
I have the data folder in .rbuildignore so it doesn't increase package size, so the vignette can't get to it at compile time